File: ideal-core/src/main/java/cn/idealframework2/cache/memory/CaffeineCacheBuilderFactory.java
package cn.idealframework2.cache.memory;

import cn.idealframework2.cache.CacheBuilder;
import cn.idealframework2.cache.CacheBuilderFactory;
import cn.idealframework2.cache.serialize.ValueSerializer;

import javax.annotation.Nonnull;

/**
 * @author 宋志宗 on 2022/12/5
 */
public class CaffeineCacheBuilderFactory implements CacheBuilderFactory {

  @Nonnull
  @Override
  public <K, V> CacheBuilder<K, V> newBuilder(@Nonnull ValueSerializer<V> valueSerializer) {
    return null;
  }
}

File: ideal-core/src/main/kotlin/cn/idealframework2/cache/coroutine/memory/CaffeineCacheBuilderFactory.kt
package cn.idealframework2.cache.coroutine.memory

import cn.idealframework2.cache.coroutine.CacheBuilder
import cn.idealframework2.cache.coroutine.CacheBuilderFactory
import cn.idealframework2.cache.serialize.ValueSerializer

/**
 * @author 宋志宗 on 2022/12/5
 */
class CaffeineCacheBuilderFactory : CacheBuilderFactory {

  override fun <K : Any, V : Any> newBuilder(valueSerializer: ValueSerializer<V>): CacheBuilder<K, V> {
    return CaffeineCacheBuilder()
  }
}
